Content.Server/StationEvents/Events/MassHallucinationsRule.cs

```diff
@@ -35,6 +36,7 @@ protected override void Ended(EntityUid uid, MassHallucinationsRuleComponent com
         while (query.MoveNext(out var ent, out _))
         {
             RemComp<ParacusiaComponent>(ent);
+            RemComp<MassHallucinationsComponent>(ent);
         }
```

Comment:
You should not be modifying collections while enumerating them as this will immediately throw exceptions.

What this event should actually do is track who it's affected and then shut itself down cleanly instead of globally making changes.